Content.Server/Chemistry/ReagentEffects/DoAction.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Content.Shared.Actions;
+using Content.Shared.Actions.Components;
+using Content.Shared.Actions.Prototypes;
+using Content.Shared.Chemistry.Reagent;
+using Content.Shared.Construction.Steps;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.Manager.Attributes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Server.Chemistry.ReagentEffects;
+
+/// <summary>
+///     Forces someone to do a certain action, if they have it.
+/// </summary>
+public class DoAction : ReagentEffect
+{
+    [DataField("action", required: true, customTypeSerializer:typeof(PrototypeIdSerializer<ActionPrototype>))]
+    public string Action = default!;
+
+    public override void Effect(ReagentEffectArgs args)
+    {
+        if (args.EntityManager.TryGetComponent(args.SolutionEntity, out SharedActionsComponent? actions))
+        {
+            if (!IoCManager.Resolve<IPrototypeManager>().TryIndex<ActionPrototype>(Action, out var proto))
+                return;
```

Comment:
Should log this as an error IMO so someone can pick it up (grafana or when we get a test for it).

Reply:
If this fails it means there's an issue with the linter / customTypeSerializer right? Surely something else would go wrong elsewhere as well.

Reply:
(In that case I would just Index it but it's probably fine).